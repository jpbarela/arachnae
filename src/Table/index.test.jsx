// @flow
import * as React from "react";
import { render } from "@testing-library/react";
import { Table, TableBody, TableData, TableHead, TableHeader, TableRow } from "./index";
import { StyleSheetTestUtils } from "aphrodite";

describe("Row", () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it("renders", () => {
    const container = render(
    <Table>
        <TableHead>
            <TableRow><TableHeader>Name</TableHeader><TableHeader>Salary</TableHeader></TableRow>
        </TableHead>
        <TableBody>
          <TableRow><TableData>Joe</TableData><TableData>$100</TableData></TableRow>
          <TableRow><TableData>Jane</TableData><TableData>$150</TableData></TableRow>
        </TableBody>
    </Table>
    );
    expect(container.baseElement.firstChild).toMatchSnapshot();
  });
});
