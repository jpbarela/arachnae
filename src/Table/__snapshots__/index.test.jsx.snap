// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders 1`] = `
<div>
  <table
    class="table_1yarp85"
  >
    <thead>
      <tr>
        <th
          class=""
          colspan="1"
          rowspan="1"
        >
          Name
        </th>
        <th
          class=""
          colspan="1"
          rowspan="1"
        >
          Salary
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="tableData_10e23z7"
          colspan="1"
          rowspan="1"
        >
          Joe
        </td>
        <td
          class="tableData_10e23z7"
          colspan="1"
          rowspan="1"
        >
          $100
        </td>
      </tr>
      <tr>
        <td
          class="tableData_10e23z7"
          colspan="1"
          rowspan="1"
        >
          Jane
        </td>
        <td
          class="tableData_10e23z7"
          colspan="1"
          rowspan="1"
        >
          $150
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
